src/mds/CDir.cc

```diff
@@ -2160,10 +2160,157 @@ class C_IO_Dir_Committed : public CDirIOContext {
     dir->_committed(r, version);
   }
   void print(ostream& out) const override {
-    out << "dirfrag_commit(" << dir->dirfrag() << ")";
+    out << "dirfrag_committed(" << dir->dirfrag() << ")";
   }
 };
 
+class C_IO_Dir_Commit_Ops : public Context {
+public:
+  C_IO_Dir_Commit_Ops(CDir *d, version_t v, int pr, set<dentry_key_t> &rm,
+                      std::map<dentry_key_t, CDir::dentry_commit_items> &s,
```

Comment:
don't use map/set. use vector with preallocate space

Reply:
Sure, will fix it.